k2/csrc/properties.cc

```diff
@@ -78,6 +79,41 @@ bool HasSelfLoops(const Fsa &fsa) {
   return false;
 }
 
+bool CheckCycles(StateId s, std::vector<bool> visited, std::vector<bool> back_arc,
```

Comment:
It would be more efficient if you pass  const references as function  arguments.

```cpp
const std::vector<bool>&
```

Reply:
done. I passed by reference, and as const when possible.

Reply:
The Google style checker gives me the following error despite I'm using `std::vector<bool> &visited`. Do I have to use pointer (*) instead of reference (&) in the function arguments?

```
Is this a non-const reference? If so, make const or use a pointer: std::vector<bool> &visited  [runtime/references]
```

Reply:
Yes. For output arguments, please use `pointer`.

Reply:
done.